addons/payment_stripe/models/payment_transaction.py

```diff
@@ -99,7 +99,7 @@ def _stripe_create_checkout_session(self):
                     'line_items[0][price_data][currency]': self.currency_id.name,
                     'line_items[0][price_data][product_data][name]': self.reference,
                     'line_items[0][price_data][unit_amount]': payment_utils.to_minor_currency_units(
-                        self.amount, self.currency_id
+                        self.amount, self.currency_id, CURRENCY_ISOCODE_EXPONENT.get(self.currency_id.name, None)
```

Comment:
This param is intended for non-ISO values. For example, and as you stated in https://github.com/odoo/odoo/pull/106130#issuecomment-1321759057, [Adyen does not follow the ISO norm for all currencies and requires different exponents for 4 of them](https://github.com/odoo/odoo/blob/15.0/addons/payment_adyen/const.py#L14-L22).

Instead, update the `to_minor_currency_units` and `to_major_currency_units` util functions to use the hardcoded ISO table rather than the `decimal_places` field.